src/setup/cortx_deploy

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+DEFAULT_LOG_LEVEL=4
+
+# Log Levels: 0-NOTSET 1-FATAL 2-ERROR 3-WARNING 4-INFO 5-DEBUG
+function log {
+    level=$1; shift
+    LOG_LEVEL=${DEBUG:-$DEFAULT_LOG_LEVEL}
+    [[ $level -gt $LOG_LEVEL ]] && return
+
+    echo "log: $*"
+}
+
+function show_usage {
+    echo -e "usage: $(basename $0) -n NODE-ID [-f SOLUTION-CONFIG] [-d LEVEL][-o CORTX-CONFIG-URL]"
+    echo -e "Where:"
+    echo -e "..."
+    echo -e " NODE-ID         : Unique Node ID"
+    echo -e " SOLUTION-CONFIG : Solution-Config directory containing - config.yaml, cluster.yaml and secrets.yaml"
+    echo -e "                   (Default: /etc/cortx/solution)"
+    echo -e " LEVEL           : Debug Log Level. By default Level is 0"
+    echo -e " CORTX-CONFIG-URL: URL of CORTX Configuration. Default is yaml:///etc/cortx/cluster.conf"
+    exit 1
+}
+
+function validate_config {
+    local config=$1
+    if [ ! -f $config ]; then
+        echo -e "ERROR: Config $config not found"
+    fi
+}
+
+function yaml_config {
+    local config=$1
+    echo "yaml://$config"
+}
+
+# Constants
+DEBUG_LOG=false
+NODE_ID=0
+SETUP_ARGS=""
+SOLUTION_CONFIG="/etc/cortx/solution"
+CORTX_CONFSTORE="/etc/cortx/cluster.conf"
+
+while [ $# -gt 0 ];  do
+    case $1 in
+    -d )
+        shift 1
+        LOG_LEVEL=$1;
+        ;;
+
+    -n )
+        shift 1
+        NODE_ID=$1
+        ;;
+
+    -m )
+        shift 1
+        SETUP_ARGS="-m"
+        ;;
+
+    -c )
+        shift 1
+        SOLUTION_CONFIG=$1
+        ;;
+
+    -o )
+        shift 1
+        CORTX_CONFSTORE=$1
+        ;;
+
+    -h )
+        show_usage
+        ;;
+
+
+    * )
+        echo "Invalid argument provided : $1"
+        show_usage
+        exit 1
+        ;;
+
+    esac
+    shift 1
+done
+
+# path for solution-config cluster.yaml and config.yaml files
+CONFIG_INFO="$SOLUTION_CONFIG/config.yaml"
+validate_config $CONFIG_INFO
+CLUSTER_INFO="$SOLUTION_CONFIG/cluster.yaml"
+validate_config $CLUSTER_INFO
+
+export PATH=$PATH:/opt/seagate/provisioner/bin
+
+[ -z $NODE_ID ] && echo -e "ERROR: Missing NODE-ID" && show_usage
+echo $NODE_ID > "/etc/machine-id";
+log $LOG_LEVEL node_id=$NODE_ID
+
+# Apply Cluster Config
+log $LOG_LEVEL "cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -o $(yaml_config $CORTX_CONFSTORE)"
```

Comment:
parameters to log should be fixed. e.g. 

```
log 5 "cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -o $(yaml_config $CORTX_CONFSTORE)"
```

or
```
function log_debug { 
    log 5 $* 
}

function log_info { 
    log 4 $* 
}

log_debug "cortx_setup config apply -f $(yaml_config $CLUSTER_INFO) -o $(yaml_config $CORTX_CONFSTORE)"
```

Reply:
done